core/cache/lxdprofilewatcher.go

```diff
@@ -148,24 +149,24 @@ func (w *MachineAppLXDProfileWatcher) unitChange(topic string, value interface{}
 		notify = w.removeUnit(names)
 	case okUnit:
 		switch {
-		case unit.details.MachineId == "" && !unit.details.Subordinate:
```

Comment:
if each call to MachineId() now needs to take out a mutex lock, it starts looking like something where we should be locally caching the machine id, since it isn't something that should ever actually be changing.
But that leads me to wonder why we are protecting it with a mutex... etc.